packages/server/src/controllers/Controller.js

```diff
@@ -111,6 +112,68 @@ export class Controller {
   }
 
   setupAction(type, name, handler, authorize) {
+    if (!isFunction(handler)) {
+      const setupHandlerFromParam = ({
+        handler,
+        action,
+        authorize,
+        parameters,
+        returns,
+        scope,
+        transacted
+      }) => {
+        if (action !== undefined) {
+          handler.verb = action.verb
+          handler.path = action.path
+        }
+
+        if (authorize !== undefined) {
+          handler.authorize = authorize
+        }
+
+        if (parameters !== undefined) {
+          const parametersHasOptions = isArray(parameters[0])
```

Comment:
How about:

```js
const [ params, options ] = asArray(parameters)
```

Reply:
This works for the case where `parameters: [[parameterA, parameterB], options]` But `parameters: [parameterA, parameterB]` would result in `[_parameters = parameterA, options = parameterB]`.

Reply:
Updated to:

```js
  if (parameters) {
    const [_parameters, options] = parameters
    const hasOptions = isArray(_parameters)
    handler.parameters = hasOptions ? _parameters : parameters

    // If validation options are provided, expose them through
    // `handler.options.parameters`, see ControllerAction
    if (hasOptions) {
      handler.options = handler.options || {}
      handler.options.parameters = options
    }
  }
```